### DISCUSSION TITLE
add track mod

### DISCUSSION THREAD

**Discussion by oudileoudi** [iluvcapra/py-ptsl#54] [Show and tell]
hello 
i've add insertion_point_position and pagination in create_new_tracks()
(i've add pagination because i've error pagination error each time i use it)

the argument for insertion point position are :

TIPoint_Unknown 	
Default value, the insertion point is determined by Pro Tools.
TIPoint_Before 	
The insertion point is before the specified track.
TIPoint_After 	
The insertion point is after the specified track.
TIPoint_First 	
The insertion point is either the first track in the session or the first member track of the specified folder track.
TIPoint_Last 	
The insertion point is either the last track in the session or the last member track of the specified folder track.

work with TIPoint_First and TIPoint_Last

sorry but i'm not an expert on code :-) 
hope it's ok for you but it's works for me.

the modification in engine.py:

`    def create_new_tracks(self,
                          number_of_tracks: Optional[int] = None,
                          track_name: Optional[str] = None,
                          track_format: Optional[TrackFormat] = None,
                          track_type: Optional[TrackType] = None,
                          track_timebase: Optional[TrackTimebase] = None,
                          insertion_point_position: Optional[insertionPointPosition] = None
                          ):
        op = ops.CreateNewTracks(number_of_tracks=number_of_tracks,
                                 track_name=track_name,
                                 track_format=track_format,
                                 track_type=track_type,
                                 track_timebase=track_timebase,
                                 pagination_request = pt.PaginationRequest(limit=1000, offset=0),
                                 insertion_point_position=  insertion_point_position,
                                 )
        """
        Create new Tracks
        """

        self.client.run(op)`

**Comment by iluvcapra** [comment]
I think this would be a good addition, in general I try to avoid using all the enumeration types in the engine interface though, would you be able to rewrite it so it took a string as the format, type, and position params and then converted these into the underlying types?

**Comment by oudileoudi** [comment]
hello 
i’m not a good coder (sorry for my English) my knowledge are really poor
but i can try :-)

i’ve add  set_track_mute_state and  set_track_solo_state if your want.
i don’t how to add in the git.
I can send by mail if you want
best from France

**Comment by oudileoudi** [comment]
def create_new_tracks(self,
                          number_of_tracks: Optional[int] = None,
                          track_name: Optional[str] = None,
                          track_format: Optional[TrackFormat] = None,
                          track_type: Optional[TrackType] = None,
                          track_timebase: Optional[TrackTimebase] = None,
                          track_Insertion_point: Optional[TrackInsertionPoint] = None
                          ):


        op = ops.CreateNewTracks(number_of_tracks=number_of_tracks,
                                 track_name=track_name,
                                 track_format=track_format,
                                 track_type=track_type,
                                 track_timebase=track_timebase,
                                 insertion_point_position =  track_Insertion_point,
                                 pagination_request=pt.PaginationRequest(limit=1000, offset=0),
                                 )

        """
        Create new Tracks
        """


        self.client.run(op)


    """
